src/plugins/dam/components/batch-edit-modal/BatchEdit.jsx

```diff
@@ -0,0 +1,193 @@
+      text,
+      title: 'Are you sure?',
+      type: 'warning',
+      reverseButtons: true,
+    });
+  }
+
+  static propTypes = {
+    assetIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+    allowMultiUpload: PropTypes.bool,
+    currentValues: PropTypes.shape({
+      credit: PropTypes.object,
+      expiresAt: PropTypes.instanceOf(Date),
+    }),
+    delegate: PropTypes.instanceOf(Delegate).isRequired,
+    getNode: PropTypes.func,
+    hasMultipleFiles: PropTypes.bool,
+    initialValues: PropTypes.shape({}),
+    isFormDirty: PropTypes.bool,
+    isFormPrestine: PropTypes.bool,
+    isFormValid: PropTypes.bool,
+    isOpen: PropTypes.bool,
+    nodeRef: PropTypes.instanceOf(NodeRef).isRequired,
+    /* eslint react/no-unused-prop-types: off */
+    onClose: PropTypes.func, // This is used in the delegate
+    onToggleBatchEdit: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    allowMultiUpload: true,
+    currentValues: {},
+    hasMultipleFiles: false,
+    getNode: noop,
+    initialValues: {},
+    isFormDirty: undefined,
+    isFormPrestine: undefined,
+    isFormValid: undefined,
+    isOpen: false,
+    onClose: noop,
+  };
+
+  constructor(props) {
+    super(props);
+    const {
+      delegate,
+    } = props;
+
+    delegate.bindToComponent(this);
+
+    this.handleToggleUploader = this.handleToggleUploader.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+  }
+
+  handleToggleUploader() {
+    const {
+      isFormDirty,
+      onToggleBatchEdit,
+    } = this.props;
+
+
+    if (isFormDirty) {
+      const confirmText = 'Do you want to leave the form without saving?';
+      BatchEdit.confirmDone(confirmText).then(({ value }) => {
+        if (value) {
+          onToggleBatchEdit();
+        } else {
+          // do nothing, user declined
+        }
+      });
+    } else {
+      onToggleBatchEdit();
+    }
+  }
+
+  handleUpdate() {
+    const {
+      assetIds,
+      currentValues,
+      delegate,
+      onToggleBatchEdit,
+    } = this.props;
+
+    // Slug is in currentValues sometimes so we must have a white list.
+    const filter = ['title', 'credit', 'expiresAt', 'description'];
+
+    const filteredCurrentValues = {};
+    Object.keys(currentValues).forEach((key) => {
+      if (currentValues[key] && filter.indexOf(key) > -1) {
+        filteredCurrentValues[key] = currentValues[key];
+      }
+    });
+
+    delegate.handleUpdate(filteredCurrentValues, assetIds, onToggleBatchEdit);
+  }
+
+  render() {
+    const {
+      assetIds,
+      nodeRef,
+      delegate,
+      currentValues,
+      getNode,
+      isFormDirty,
+      isFormValid,
+      isFormPrestine,
+      isOpen,
+    } = this.props;
+
+    return (
+      <Modal isOpen={isOpen} toggle={this.handleToggleUploader} size="md">
+        <ModalHeader toggle={this.handleToggleUploader}>
+          Batch Edit
+        </ModalHeader>
+        <ModalBody>
+          <Card>
+            <Form
```

Comment:
@rsumilang far as i can tell the only props this really needs is `form` and `initialValues` (and initialValues is a stretch because the form is not getting hydrated, even if only a single asset is selected). also no validation is happening? this is kind of one foot in redux form and one foot in regular react - for example this doesnt need `onSave` or `onSubmit` because the form is never being submitted, its just a click handler on the button